### DISCUSSION TITLE
Denormalize array into entity with DateTimeImmutable

### DISCUSSION THREAD

**Discussion by fabienlem** [symfony/symfony#48770] [Q&A]
### Symfony version(s) affected

6.2

### Description

I'm using the serializer to denormalize an array into a Doctrine entity with some DateTimeImmutable fields.

I get the following error: "Expected argument of type "?DateTimeImmutable", "DateTime" given ".

Is there a way to configure the date denormalizer to use DataTimeImmuable instead of DateTime when date denormalization is nested?

### How to reproduce

```
class MyDoctrineEntity {
    #[ORM\Column(type: 'datetimetz', nullable: true)]
    private ?\DateTimeImmutable $importedAt = null;
}

/** @var SerializerInterface $serializer */
$serializer->denormalize(['importedAt' => '2020-12-10 09:35:35.050'], MyDoctrineEntity::class);
```

### Additional Context

I can't find how to achieve this in the serializer documentation.

**Comment by derrabus** [comment]
The Doctrine type `datetimez` maps to `DateTime`, not `DateTimeImmutable`, see https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/types.html#datetimetz
